source.lua

```diff
@@ -54,31 +84,55 @@ function AddLiquidSource(pos, liquid, amount)
             end
         end
     end)
+    persist_state(sources)
 end
 
-function DeleteLiquidSource(pos)
-    for k, v in pairs(liquidSources) do
-        if same_xyz(pos, v.pos) then liquidSources[k] = nil end
-        return
+function delete_source_at(key)
+    local sources = retrieve_state()
+    local v = sources[key]
+
+    if v then
+        local block = dfhack.maps.getTileBlock(v.pos)
+        if block and is_flow_passable(v.pos) then
+            local flags = dfhack.maps.getTileFlags(v.pos)
+            flags.flow_size = 0
+            dfhack.maps.enableBlockUpdates(block, true)
+        end
+        sources[key] = nil
+    end
+    persist_state(sources)
+    load_liquid_source()
+end
+
+function delete_liquid_source(pos)
+    print(("Deleting Source at [%d, %d, %d]"):format(pos.x, pos.y, pos.z))
+    for k, v in pairs(retrieve_state()) do
```

Comment:
since you're using `table.insert` to populate the table, it's a list, not a map. using `ipairs` instead of `pairs` here (and elsewhere in this file) would be semantically clearer.